raiden/utils/upgrades.py

```diff
@@ -109,93 +126,78 @@ class UpgradeManager:
 
-                # The version number in the database is smaller then the
-                # current target, this means the migration failed to execute on
-                # the last iteration, delete the partially upgraded database
-                # and start again.
-                self._delete_current_db()
+        # The latest version matches our target version, nothing to do.
+        if file_version == RAIDEN_DB_VERSION:
+            return
 
-            older_version = get_db_version(old_db_filename)
-            if not older_version:
-                # There are no older versions to upgrade from.
-                return
+        if file_version > RAIDEN_DB_VERSION:
+            raise RuntimeError(
+                f'Conflicting database versions detected, latest db version is {file_version}, '
+                f'Raiden client version is {RAIDEN_DB_VERSION}.'
+                f'\n\n'
+                f'Running a downgraded version of Raiden after an upgrade is not supported, '
+                f'because the transfers done with the new client are not understandable by the '
+                f'older.',
+            )
+
+        self._upgrade(
+            target_file=str(self._current_db_filename),
+            from_file=latest_db_path,
+            from_version=file_version,
+        )
 
-            _copy(str(old_db_filename), str(self._current_db_filename))
+    def _upgrade(self, target_file: str, from_file: str, from_version: int):
+        with get_file_lock(from_file), get_file_lock(target_file):
+            _copy(from_file, target_file)
 
-            storage = SQLiteStorage(str(self._current_db_filename))
+            storage = SQLiteStorage(target_file)
 
-            log.debug(f'Upgrading database from {older_version} to v{RAIDEN_DB_VERSION}')
+            log.debug(f'Upgrading database from {from_version} to {RAIDEN_DB_VERSION}')
 
             try:
                 with storage.transaction():
-                    version_iteration = older_version
+                    version_iteration = from_version
                     for upgrade_func in UPGRADES_LIST:
-                        extra_args = {'web3': self._web3}
-
-                        version_iteration = _run_upgrade_func(
-                            storage,
-                            upgrade_func,
-                            version_iteration,
-                            **extra_args,
+                        version_iteration = upgrade_func(
+                            storage=storage,
+                            old_version=version_iteration,
+                            current_version=RAIDEN_DB_VERSION,
+                            web3=self._web3,
                         )
 
                     update_version(storage, RAIDEN_DB_VERSION)
-                    # Prevent the upgrade from happening on next restart
-                    _backup_old_db(str(old_db_filename))
             except Exception as e:
-                self._delete_current_db()
```

Comment:
with a SIGKILL this can be skipped, so instead of having a delete at the start and at the end of the function, I just moved everything to the top.